### ISSUE TITLE
Enable all endpoint and ports for HTTPS

### ISSUE THREAD

**Issue by ag26jan** [fritshoogland-yugabyte/yb_stats#6]
Hey Frits, 

Hope you are doing good. Not sure you are still active and keeping a tab on this project. 

Recently we found that the endpoint is not TLS enabled (if YBDB universe is). Hence the `yb_stats` failed to collect the data over HTTPS. 

```
yb_stats --hosts `hostname -i` --ports 7000,9000,12000,13000,9300

[2023-10-09T14:19:37Z ERROR yb_stats::metrics] Fatal: error reading from URL: error sending request for url (http://<IP>:13000/metrics): error trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915: (unable to get local issuer certificate)
```

Here are the ports which is now listening on HTTPS. 

```
./port_check.sh
Checking port 7000: SSL (HTTPS)
Checking port 9000: SSL (HTTPS)
Checking port 12000: SSL (HTTPS)
Checking port 13000: SSL (HTTPS)
Checking port 9300: Non-SSL (HTTP)

OR 

openssl s_client -connect `hostname -i`:13000
CONNECTED(00000003)
Can't use SSL_get_servername
depth=0 CN = x.x.x.x
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = x.x.x.x
verify error:num=21:unable to verify the first certificate
verify return:1
depth=0 CN = x.x.x.x
verify return:1
---
Certificate chain
 0 s:CN = x.x.x.x
   i:CN = yb-dev-agangwar-test, O = example.com
---
Server certificate
-----BEGIN CERTIFICATE-----

```

Another proof is:

```
[centos@ip-10-9-127-46 ~]$ curl -s http://x.x.x.x:7000/api/v1/tablet-servers | jq
[centos@ip-10-9-127-46 ~]$ curl -sk https://x.x.x.x:7000/api/v1/tablet-servers | jq
{
  "7a730b6a-9e1f-4d92-b03a-8d10ad03e356": {
    "x.x.x.x:9000": {
      "time_since_hb": "0.7s",
      "time_since_hb_sec": 0.668659675,
      "status": "ALIVE",
      "uptime_seconds": 21036,
      "ram_used": "70.80 MB",
      "ram_used_bytes": 70803456,
      "num_sst_files": 0,
      "total_sst_file_size": "0 B",
      "total_sst_file_size_bytes": 0,
      "uncompressed_sst_file_size": "0 B",
      "uncompressed_sst_file_size_bytes": 0,
      "path_metrics": [
        {
          "path": "/mnt/d0",
          "space_used": 1951703040,
          "total_space_size": 268304384000
        }
      ],
      "read_ops_per_sec": 0.0,
      "write_ops_per_sec": 0.0,
      "user_tablets_total": 1,
      "user_tablets_leaders": 1,
      "system_tablets_total": 4,
      "system_tablets_leaders": 4,
      "active_tablets": 5,
      "cloud": "aws",
      "region": "us-west-2",
      "zone": "us-west-2a"
    }
  }
}
```

Can we do this quick fix.

**Comment by ag26jan** [comment]
I tried to fix this per below but no luck.

```
pub fn http_get(
    host: &str,
    port: &str,
    url: &str,
) -> String
{
    let url_string = format!("http://{}:{}/{}", host, port, url);
    let url = Url::parse(&url_string).unwrap();

    let client = Client::builder()
        .connect_timeout(Duration::from_millis(200))
        .danger_accept_invalid_certs(ACCEPT_INVALID_CERTS)
        .build()
        .unwrap();

    let request = client.get(url);

    // Add headers if needed
    // request.header("Header-Name", HeaderValue::from_static("Header-Value"));

    if let Ok(data_from_web_request) = request.send() {
        if !data_from_web_request.status().is_success() {
            debug!(
                "Non success response: {}",
                data_from_web_request.status()
            );
        } else {
            debug!(
                "Success response: {}",
                data_from_web_request.status()
            );
        }
        data_from_web_request.text().unwrap()
    } else {
        debug!(
            "Non-Ok success response: {}",
            url_string
        );
        String::new()
    }
}

**Comment by FritsHoogland** [comment]
Solved with version 0.9.8.
